oeps/oep-0050-hooks-extension-framework.rst

```diff
@@ -0,0 +1,227 @@
+===================================
+OEP-0050: Hooks extension framework
+===================================
+
+.. list-table::
+   :widths: 25 75
+
+   * - OEP
+     - :doc:`OEP-0050 </oeps/oep-0050-arch-hook-based-extensions>`
+   * - Title
+     - Open edX Hooks Extension Framework
+   * - Last Modified
+     - 2021-02-04
+   * - Authors
+     - Felipe Montoya <felipe.montoya@edunext.co>
+   * - Arbiter
+     - < TBD >
+   * - Status
+     - Draft
+   * - Type
+     - Architecture
+   * - Created
+     - 2021-02-04
+   * - Review Period
+     - < TBD >
+
+Abstract
+========
+
+To sustain the growth of the Open edX ecosystem, the business rules of the
+platform must be open for extension following the open-closed principle. This
+OEP specifies a solution using what is commonly referred to as hooking. That is
+the definition of a stable set of places in the code (triggers) where externally
+defined actions (via plugins) can take place.
+
+
+Motivation
+==========
+
+Following the open-closed principle, it would be desirable for the open edX
+project to allow the extension points (great `summary of the current options`_)
+to modify as much of the code paths that define the business rules of an
+open edX instance in a way that is consistent and repeatable.
+
+Since the creation of the plugin framework different initiatives in the
+community have demonstrated success by adopting this pattern. This OEP extends
+the range of possible extensions that the patter can hold sustainably.
+
+Adopters of the Open edX technology as a way of delivering online courses very
+often require that certain changes are made to the codebase by changing the
+edx-platform core. This proposal aims to be a solution to organize the majority
+of the changes that the authors have encountered after years of work extending
+the core in a way that is more compatible with the community at large, it is
+easier to reuse and far easier to migrate to newer versions of the codebase.
+
+This proposal draws inspiration from the cumulative experience of eduNEXT as the
+lead author, other members of the community, and from one of the largest
+extendable platforms for the web today for many years already, WordPress.
+
+.. _summary of the current options: https://github.com/edx/edx-platform/blob/master/docs/guides/extension_points.rst
```

Comment:
Just a heads up that this file may be getting moved so the URL will eventually need updating - see https://github.com/edx/edx-developer-docs/pull/37

Reply:
Thanks for making me aware of this. I understand that the effort of the developers docs is a bit in a hiatus at the moment. In any case I'll be looking at this to maintain the documentation in the correct place.